### DISCUSSION TITLE
Supports Nested Pages(NuxtChild)

### DISCUSSION THREAD

**Discussion by dadajam4** [nuxt/nuxt#18348] [Questions] [answered]
I used to use NuxtChild as a very useful feature of Nuxt2.
https://nuxtjs.org/examples/routing/nested-pages/

Now, in Nuxt3, I haven't found an equivalent feature to NuxtChild.
I would like to know if there are plans to include similar features in the future, or if they already exist but are just not documented.

I am very excited about the development of Nuxt3. Thank you.

**Comment by lucpotage** [comment]
Did you try `<slot/>` instead of `<NuxtChild/>`?

**Reply by JDIZM** [reply]
I think it's the routing of a nested child component that's causing issues/confusion with the addition of `<NuxtPage /> `

The goal is to achieve nested routes like this https://router.vuejs.org/guide/essentials/nested-routes.html

I was thinking maybe teleport would be useful https://v3.vuejs.org/guide/teleport.html#using-with-vue-components

The above solution may work, just adding some thoughts.

**Reply by dadajam4** [reply]
@imlautaro 
Thanks.. But
I also tried <slot/>, but that didn't solve the problem. Looking at the error log, I believe that it is failing to generate the routing in the first place.

```
[Vue Router warn]: Component "default" in record with path "/page3" is not a valid component. Received "undefined".
[Vue Router warn]: uncaught error during route navigation:
Error: Invalid route component
    at extractComponentsGuards (/Users/dadajam4/projects/develops/nuxt3-app/node_modules/vue-router/dist/vue-router.cjs.js:1973:27)
```
This problem seems to occur when a vue file with the same name as the directory is placed under the pages directory.

````
pages/
  page3/
    index.vue
  page3.vue
```

**Reply by lucpotage** [reply]
You’re right, it does not seem implemented yet. I don’t know if it will.

**Comment by TenshiGoan** [comment]
When i named component and dir "sub", in "./.nuxt/routes.mjs" for ./sub/index.js don't exist component/
![image](https://user-images.githubusercontent.com/471072/137122700-e77d23ec-1ff1-46d4-bfa7-9d513691346e.png)

**Comment by nathanchase** [comment, marked as answer]
@dadajam4 @lucpotage @TenshiGoan @JDIZM @dadajam4 I believe that `<NuxtPage>` is the replacement for both `<nuxt>` AND `<nuxt-child>`. See @Atinux's slides from this video: https://www.youtube.com/watch?v=LMONTMOc1zI&t=126s
![image](https://user-images.githubusercontent.com/5963574/137126714-372bf69f-4c2e-4aa6-8571-a43b74bfface.png)

**Reply by TenshiGoan** [reply]
we know about nuxt-child become nuxt-page. as is i see(can be wrong) problem with config generated for vue-router. later i try to see(now i at work) what code generete it, and what vue-router needed for it to work.

**Comment by TenshiGoan** [comment]
found why component field don't created, There added component only to top routes: 
https://github.com/nuxt/framework/blob/d8332d6ad6744c0e11ff6d3eff7138d3aae822c1/packages/nuxt3/src/pages/module.ts#L43

**Reply by TenshiGoan** [reply]
it's start working when i replesed this code with:
```
const map_routes = routes => routes.map((route) => ({...route, children: map_routes(route.children), component: `{() => import('${route.file}')}` }));
const serializedRoutes = map_routes(routes);

```

**Comment by dadajam4** [comment]
@nathanchase 
@TenshiGoan 
Very thanks!
You made it at I could understand at this is bug and need fix probrem.
I am going to report this issue as an issue.

**Reply by dadajam4** [reply]
Reported as an issue.
https://github.com/nuxt/framework/issues/963
